src/flamegpu/model/SubEnvironmentDescription.cpp

```diff
@@ -63,28 +63,90 @@ void SubEnvironmentDescription::mapProperty(const std::string &sub_property_name
     // Properties match, create mapping
     data->properties.emplace(sub_property_name, master_property_name);
 }
+void SubEnvironmentDescription::mapMacroProperty(const std::string& sub_property_name, const std::string& master_property_name) {
+    // Sub macro property exists
+    auto subEnv = data->subEnvironment.lock();
+    if (!subEnv) {
+        THROW exception::InvalidParent("SubEnvironment pointer has expired, "
+            "in SubEnvironmentDescription::mapMacroProperty()\n");
+    }
+    const auto subProp = subEnv->macro_properties.find(sub_property_name);
+    if (subProp == subEnv->macro_properties.end()) {
+        const auto parent = data->parent.lock();
+        THROW exception::InvalidEnvProperty("SubModel '%s's Environment does not contain property '%s', "
```

Comment:
Should this mention macro early in the exception message? (as in exceptions below?)